server/src/main/java/com/cloud/storage/VolumeApiServiceImpl.java

```diff
@@ -1187,6 +1189,16 @@ private VolumeVO orchestrateResizeVolume(long volumeId, long currentSize, long n
             if (currentSize != newSize && _volsDao.getHypervisorType(volume.getId()) == HypervisorType.XenServer && !userVm.getState().equals(State.Stopped)) {
                 throw new InvalidParameterValueException(errorMsg);
             }
+
+            /* Do not resize volume of running vm on KVM host if host is not Up or not Enabled */
+            if (currentSize != newSize && userVm.getState() == State.Running && userVm.getHypervisorType() == HypervisorType.KVM) {
+                HostVO host = _hostDao.findById(userVm.getHostId());
+                if (host.getStatus() != Status.Up) {
```

Comment:
@ustcweizhou should you do a null check, what if the host was removed?

Reply:
@rhtyd added null check